### DISCUSSION TITLE
Table list view - column with icon and HTML tooltip

### DISCUSSION THREAD

**Discussion by craigcordell** [filamentphp/filament#14171] [Showcase]
// show notes as an icon with an HTML tooltip
                TextColumn::make('notes')
                    ->alignCenter()
                    ->searchable()

                   // show an icon if we have notes
                    ->icon(function (string|null $state) {
                        return ($state) ? 'heroicon-o-information-circle' : null;
                    })
                    ->color(function (string|null $state) {
                        return ($state) ? 'success' : null;
                    })

                    // the rest of these are critical to getting the tooltip to render as HTML
                    ->html()
                    // return empty string if you just want an icon
                    ->formatStateUsing(function ($state) {
                        return ''; 
                    })
                    // I'm just rendering the notes but you have access to the record
                    ->extraAttributes(fn(Model $record): array => [ 
                        'x-tooltip.html' => new HtmlString(),
                        'x-tooltip.raw' => new HtmlString(nl2br($record->notes)),
                    ]),

**Comment by paanblogger** [comment]
```
IconColumn::make('notes')
->icon('heroicon-o-information-circle')
->label(false)
->color('info')
->extraAttributes(fn(Model $record): array => [
'x-tooltip.html' => new HtmlString(),
'x-tooltip.raw' => new HtmlString(nl2br($record->notes )),
])
```

**Reply by sinnbeck** [reply]
@craigcordell Awesome. Just what I needed :) 
@micraux That just renders the  `<br>` tags?

**Reply by micraux** [reply]
@sinnbeck Yes, sorry, my mistake: I mixed up two functions from my code...

I just deleted my comment...
